### DISCUSSION TITLE
[PostgreSQL] How to set bigserial id instead of serial id?

### DISCUSSION THREAD

**Discussion by MaxmaxmaximusGitHub** [prisma/prisma#2486] [Q&A] [answered]
[PostgreSQL] How to set `bigserial` id instead of `serial` id?

this create serial id:
```
model User {
  id    Int @id @default(autoincrement())
  name  String
}
```

this not works
```
model User {
  id    bigserial @id @default(autoincrement())
  name  String
}
```

i use
```sh
 prisma migrate save --experimental
 prisma migrate up --experimental
```

**Comment by beeplin** [comment]
I dont think prisma is able to handle this right now. The only way might be going into your db and doing a `alter table` manually.

**Reply by MaxmaxmaximusGitHub** [reply]
I think for this you need to add directives =) tooltips for migration tool, like this:


```prisma
model User {
  id    Int @id @default(autoincrement()) @type(bigserial)
  name  String
}
```

and if the Prisma field can be implemented by several types of fields in the database, then the programmer will be able to specify the preferred type

**Comment by zacwebb** [comment, marked as answer]
I know this is over a year old but this question comes up as the first Google result...  
with [2.17.0](https://github.com/prisma/prisma/releases/tag/2.17.0) you should be able to use [native types](https://www.prisma.io/docs/concepts/components/prisma-schema/data-model#native-types-mapping) to implement a `BIGINT` autoincrement ID:

```
Int @db.BigInt @default(autoincrement())
```

Haven't actually tested this but going by the release notes this should work

**Reply by ryands17** [reply]
Yes this will work 👍

**Comment by ludralph** [comment]
Hi @MaxmaxmaximusGitHub,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
